module/VuFindSearch/src/VuFindSearch/Backend/EDS/Base.php

```diff
@@ -41,12 +43,16 @@
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     http://edswiki.ebscohost.com/EDS_API_Documentation
  */
-abstract class Base
+abstract class Base implements LoggerAwareInterface
 {
+    use \VuFind\Log\LoggerAwareTrait { debug as public debugTrait; }
+
     /**
      * A boolean value determining whether to print debug information
      *
      * @var bool
+     *
+     * @deprecated Set the logging level in the Logging section of config.ini.
      */
     protected $debug = false;
```

Comment:
Because of keeping the $debug boolean around (if deprecated), I can't just use LoggerAwareTrait this->debug() as is.  `debugTrait` is a ridiculous function name but we can remove it next month for 10.0.

Reply:
Ooh, that is annoying! I'd be tempted to fudge a little bit and just chop out the $debug property now. It's a lot more likely that somebody is calling debugPrint in a local subclass than that they're changing the way that property gets assigned. Otherwise, the `@deprecated` comment in this file isn't really correct, and we have to do significant re-rewriting in 10.0. I'd be comfortable putting a changelog note about the removed property and deprecation in 9.1 unless you think that's terribly irresponsible. ;-)

Reply:
...and while I say "a lot more likely" I actually think it's extremely unlikely that anybody is extending any of this. :-)

Reply:
Yeah ok I agree 100%.  Will cut out that property and restore some sanity to the function name.

Reply:
I'm nearly out of time for today, so I'll probably get this merged next week, assuming you finish it up tonight. (No rush in any case).

Reply:
I have a second logging PR to do anyway.  No rush.